### ISSUE TITLE
500 Server Error on macOS trying to access processes endpoint

### ISSUE THREAD

**Issue by Endareth** [NagiosEnterprises/ncpa#493]
I'm running Nagios Core 4.4.2, and trying to monitor processes on another Mac using ncpa 2.1.5, using check_ncpa.py. Most of the standard checks (cpu/memory/disk) are working fine for me, and the process checks work fine for my Windows machines, but on the Mac, trying to run:
`check_ncpa.py -H build-s1 -t '<redacted>' -P 5693 -M 'processes' -q 'search=distnoted' -v`
I'm getting back:
`An error occurred:HTTP Error 500: INTERNAL SERVER ERROR`

Looking in the listener logs on the ncpa client, I've got the following error details:
```
2018-10-04 16:32:20,576 66720 ERROR Exception on /api/processes/ [GET]
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
  File "/Users/ncpa/ncpa/agent/listener/server.py", line 169, in token_auth_decoration
  File "/Users/ncpa/ncpa/agent/listener/server.py", line 927, in api
  File "/Users/ncpa/ncpa/agent/listener/processes.py", line 402, in run_check
TypeError: unsupported operand type(s) for +=: 'int' and 'str'
2018-10-04 16:32:20,578 66720 INFO 192.168.30.164 - - [2018-10-04 16:32:20] "GET /api/processes/?token=<redacted>&check=1&search=distnoted HTTP/1.1" 500 2362 0.111640
```

I *am* able to get a straight process list using the following (which includes details of the `distnoted` I was testing with above:
`check_ncpa.py -H build-s1 -t '<redacted>' -P 5693 -M 'processes' -l`
With the following log details:
`2018-10-04 16:33:58,107 66720 INFO 192.168.30.164 - - [2018-10-04 16:33:58] "GET /api/processes/?token=<redacted> HTTP/1.1" 200 178060 0.125664`

So from a cursory look at the `processes.py` source, it looks like `proc['mem_percent'][0]` element is coming over as a string representation of the percentage numeric.

Oh, and the process list outputs the following details for that process, which seems to confirm that:
```
        {
            "username": "<redacted>", 
            "mem_percent": [
                "0.0", 
                "%"
            ], 
            "exe": "/usr/sbin/distnoted", 
            "name": "distnoted", 
            "cpu_percent": [
                "0.0", 
                "%"
            ], 
            "cmd": "Unknown", 
            "pid": 84373, 
            "mem_vms": [
                0, 
                "B"
            ], 
            "mem_rss": [
                0, 
                "B"
            ]
        }
```

**Comment by jomann09** [comment]
I think you want to do this:
```
&check=1&name=distnoted&match=search
```
That would search for anything with name that has `distnoted` in it, you can also change `match` to either `search` or `exact`

**Comment by Endareth** [comment]
Nope, adding `&match=search` makes no difference. The problem isn't in finding the process, the problem is in how the response is parsed.

**Comment by jomann09** [comment]
What version of macOS are you running this on?

**Comment by Endareth** [comment]
macOS 10.13.6 on both machines.

**Comment by jomann09** [comment]
Thanks for bringing this up! It's definitely a bug, having tested this on some systems here. I will fix this for 2.1.6 which should be out very soon.

**Comment by jomann09** [comment]
Closing this, fixed in d7fcaef64e10b05fdaddc64c550859e1c66fc3c2
